### PR TITLE
Add utf8Decode to support Arabic language

### PR DESCRIPTION
**Pull request by yegordovganich**
https://trello.com/c/BCWznbXV/6441-module-filter-is-not-working-on-educator

Added utf8Decode which is based on urlDecode0 but just getCharCodeAt instead of extractPercentHex as Asger suggested, works perfectly.

`
import runtime;
import formats/base64;
import formats/html/html;

main() {
	arabic = "أساسيات مراكز الاتصال";
	// println(arabic);
	expanded = expandUtf8(arabic);
	encoded = encodeBase64(expanded);
	decoded = decodeBase64(encoded);
	result = utf8Decode(decoded);
	println(arabic == result);

	quit(0);
}
`

Prints "true" in the console.

### CODE REVIEW

**Comment by alstrup** [comment]
I would prefer just to copy the entire thing to a new text/utf8.flow file and not change urlDecode.

File: lib/formats/html/html.flow [hunk 1]
```diff
@@ -490,67 +494,91 @@ urlEncode(s){
...
-					Pair((hexcode - 224) * 4096
-						+ (secondByte.first - 128) * 64
-						+ (thirdByte.first - 128),
-						thirdByte.second
-					)
-				}
-			} else if (hexcode <= 0xF7) { // 4 bytes
-				secondByte = extractPercentHex(percentPair.second);
-				thirdByte = extractPercentHex(secondByte.second);
-				fourthByte = extractPercentHex(secondByte.second);
```
Comment (invisicat) [inline lib/formats/html/html.flow]: btw shouldn't it be fourthByte = extractPercentHex(thirdByte.second); ? this implies a change in derived code too

**Comment by invisicat** [comment]
mine's here #360

File: lib/formats/html/html.flow (hunk 1 above)
Comment (yegordovganich) [reply lib/formats/html/html.flow]: sure, this is a bug at initial code, fixed, thx

**Comment by yegordovganich** [comment]
@alstrup what is the reason of copy/pasting code? In fact, I didn't touch urlDecode but just refactored it to reuse common part.
Let's move this utf8-related stuff to a separate module and import it at html.flow, OK?

**Comment by alstrup** [comment]
For this code, having a simple code is much easier to read and understand, compared to the "generalized" version which is hard to understand. So I think the copy is warranted when it is this little code that is unlikely to need to change.

**Comment by yegordovganich** [comment]
Let me not agree with you. I see what you mean about increasing complexity but we always sacrifice simplicity against clean code reducing potential bugs. I don't think I can convince you but I've just seen the obvious bug Cyril found at initial code and hopefully he fixed it in both versions. But when he copy-pasted code and modified it he added a new bug https://github.com/area9innovation/flow9/pull/360/files#r323566984 and I don't blame him, it always happens when you copy-pasting, that's why I hate it. Who knows how many bugs more were added to copy-pasted code which were not noticeable at the first glance.

**Comment by yegordovganich** [comment]
I don't want to waste your time here, but that's my opinion. You decide:)

**Comment by alstrup** [comment]
I claim he found the bug because his code was simpler.

**Pull request closed without merging**